### DISCUSSION TITLE
Same wallet in my flutter app and phantom wallet

### DISCUSSION THREAD

**Discussion by sistemaseltigre** [phantom/docs#83] [General]
guys, i have a problem with my code to generate polygon and eth wallet, im using web3dart at flutter to generate a new wallet and in my apps it's work, but when i try to get the same publickey wallets in phantom wallet importin seed phrase i get differents publickeys, u can help me with this? im generating a new wallet to eth with this code:

`
@override
  Future<Ethereum> getWallet() async {
    try {
      final random = Random.secure();
      EthereumAddress address;
      EtherAmount amount;

      configureClient();
      await storage.ready;

      EthPrivateKey privateKey;

      final mnemonicStorage = await storage.getItem('mnemonic');

      if (mnemonicStorage != null) {
        final seedPhrase = bip39.mnemonicToSeed(mnemonicStorage.toString());
        privateKey = EthPrivateKey(seedPhrase);
      } else {
        final mnemonic = bip39.generateMnemonic();
        final seedPhrase = bip39.mnemonicToSeed(mnemonic);
        privateKey = EthPrivateKey(seedPhrase);
        await storage.setItem('mnemonic', mnemonic);
      }
      final seedPhrase = bip39.mnemonicToSeed(mnemonicStorage.toString());
      EthPrivateKey privateKey = EthPrivateKey(seedPhrase);

      if (mnemonicStorage == null) {
        final mnemonic = bip39.generateMnemonic();
        mnemonicStorage = mnemonic;

        final seedPhrase = bip39.mnemonicToSeed(mnemonic);
        privateKey = EthPrivateKey(seedPhrase);
        await storage.setItem('mnemonic', mnemonic);
      }

      address = privateKey.address;
      amount = (await client.getBalance(address));

      return Ethereum(
          address: address.toString(), balance: amount.getInEther.toString());
    } catch (e) {
      return Ethereum(address: '', balance: '0');
    }
  }
`

for example my mnemonic is:
shiver exact obey noodle shaft finish view loan hurt found letter menu

**Comment by adamdelphantom** [comment]
Hey @sistemaseltigre,

Thanks for writing in!

It appears you are using BIP39 as a derivation path to generate your mnemonic. Phantom uses the BIP44 derivation path. I would recommend checking to see if the library you're using has that as an option.

https://help.phantom.app/hc/en-us/articles/12988493966227-What-derivation-paths-does-Phantom-wallet-support-

**Comment by sistemaseltigre** [comment]
Guys using Flutter/dart

[](https://github.com/xclud/web3dart/issues/25)

` 
import 'package:bip39/bip39.dart' as bip39;
import 'package:bitcoin_flutter/bitcoin_flutter.dart';

final normalisedSeedPhrase = this._seedPhraseNormalise(seedPhrase);
final String cryptMnemonic = bip39.mnemonicToEntropy(normalisedSeedPhrase);
final String privateKey = this.getPrivateKey(normalisedSeedPhrase: normalisedSeedPhrase);
final address = await this.getPublicAddress(privateKey);
debugPrint("pubKey = $address");
debugPrint("privKey = $privateKey"); 
`


`
String getPrivateKey({required String normalisedSeedPhrase}) {
    var seed = bip39.mnemonicToSeed(normalisedSeedPhrase);
    final hdWallet = new HDWallet.fromSeed(seed).derivePath("m/44'/60'/0'/0/0");
    final String? privKey = hdWallet.privKey;
    assert(privKey != null);
    return privKey ?? "";
 }
  
String _seedPhraseNormalise(String seedPhrase) {
    return _mnemonicWords(seedPhrase).join(" ");
 }

 List<String> _mnemonicWords(String mnemonic) {
    return mnemonic
        .split(" ")
        .where((item) => item.trim().isNotEmpty)
        .map((item) => item.trim())
        .toList();
 }
  
Future<EthereumAddress> getPublicAddress(String privateKey) async {
    final private = EthPrivateKey.fromHex(privateKey);
    final address = await private.extractAddress();
    return address;
}
`
